src/app/getting-started/getting-started.component.ts

```diff
@@ -47,9 +57,20 @@ export class GettingStartedComponent implements OnDestroy, OnInit {
       private providerService: ProviderService,
       private notifications: Notifications,
       private router: Router,
+      @Inject(AUTH_API_URL) apiUrl: string,
       private userService: UserService) {
 
+    // will be used to connect OpenShift Online.
+    this.oldLinkingServer = apiUrl;
+
+    // will be used in the non-angular html FORM's ACTION during the POST request
+    // to connect GitHub
+    this.linkingApiURL = apiUrl + 'token/link';
+
+    this.redirect = window.location.origin + "/_gettingstarted?wait=true";     
```

Comment:
All this new code appears to support a form submit. Why is a form submit preferable to using the existing provider service to generate requests? 

We created that service to keep that code out of the getting started component and keep the HTML relatively clean. For example, we don't need to hide the access token and redirect URL in the HTML template.

Reply:
Thanks for the review @dlabrecq - I updated the description above to show the use-cases this change is handling.

> All this new code appears to support a form submit. Why is a form submit preferable to using the existing provider service to generate requests?

The new account linking API gets rid of the need to pass `clientSession` & `sessionState` info for linking. We've simplified the API by requiring only the access token to be passed. Passing the access token in a GET request might break things since the access token is long. That's why we need the form POST.

> For example, we don't need to hide the access token and redirect URL in the HTML template.
Agreed, however we need it in the HTML because of scenario (3) in the description above.
In this scenario the user is trying to connect OSO and Github together. So, the new github is linked by a form POST and the old OSO is linked by the redirect after connecting github.

Reply:
Thanks. Makes more sense now. 

Although, we could create a simple POST request using code (from codebases.service) similar to this:

this.http
&nbsp;&nbsp;.post(url, payload, { headers: this.headers })
&nbsp;&nbsp;.map(response => {
&nbsp;&nbsp;return response.json().data as Codebase;
&nbsp;&nbsp;});

That said, the getting started page is going away soon. Not sure these details will matter? However, it would be preferable to have this functionality in a service, so we can call it from any page without a lot of setup in the HTML template -- probably from the user profile.

Reply:
Thanks @dlabrecq ,
That was the initial attempt to keep the component html off the excessive code but the issue is that the POST request returns a 303 which gets intercepted ( & rejected by the browser because its a redirect to github ). So that's where I realized than an AJAX POST with a 303 redirect can only be done using an html form post.

https://stackoverflow.com/questions/19952518/303-redirection-not-working-with-angular-http-post